### ISSUE TITLE
is it possible to store/retrive comments ?

### ISSUE THREAD

**Issue by chipitsine** [haproxytech/dataplaneapi#71]
we use free style comments to store ownership information like

`frontend ssl_vip # owner: example@example.com`

is it possible to store/retrive such comments via dataplane api ?

**Comment by mjuraga** [comment]
No, currently API does not have support for comments. The underlying config-parser however does, and this is a feature we discussed, but is currently low on our feature priorities.

**Comment by mjuraga** [comment]
Hi, a bit of an old issue, but now we added a partial support for comments, but in a more structured way. Data Plane API will now support only inline comments on the lines which are API resources (all the sections, and configuration lines inside a section which are represented as API resources, e.g. servers, binds, acls, http-requests etc...). The way they will be represented on the API as a `metadata` field on those resources, which is a json object so you can insert any key value there.

How it works is it will read the comment and if it isn't already a json object in the comment, it will create it and serialize it back as a json.

e.g.

```
frontend ssl_vip # my nice frontend
```

Will be serialized on the frontend as:

```
"metadata": {
  "comment": "my nice frontend"
}
```

And written back as:

```
frontend ssl_vip # { "comment": "my nice frontend"}
```

If it's already a json in a comment it will serialize it as is, so you could write something like:

```
frontend ssl_vip # {"owner": "example@example.com"}
```

And it will be shown as

```
"metadata": {
  "owner": "example@example.com"
}
```
